pyo3-macros-backend/build.rs

```diff
@@ -0,0 +1,14 @@
+use pyo3_build_config::pyo3_build_script_impl::{errors::Result, resolve_interpreter_config};
+
+fn configure_pyo3() -> Result<()> {
+    let interpreter_config = resolve_interpreter_config()?;
```

Comment:
If I remember correctly, using `resolve_interpreter_config` from a proc-macro crate won't work correctly when cross compiling, because the environment variables provided by cargo are for the host system, not the cross-compile target.

So I don't think we can add this build script and use `#[cfg]` in `pyo3_build_config`. Instead in `pyo3_build_config` we should use the runtime APIs, like in https://github.com/PyO3/pyo3/blob/2503a2dd5e9d8b703d3b9c6ce3ce12ce0126affe/pyo3-macros-backend/src/method.rs#L210-L215

Sorry that I did not document this 😭

Reply:
I thought of using it but I was pretty sure I can't do ifs and logic when defining globals? I am still pretty confident?

Reply:
Agreed, I think you'll need to define both globals and then check which one to use based on the Python version later in the macro code.

E.g. `__IPOW__` and `__IPOW_PY_3_7__`, and then check which one to use at runtime.

This approach might not be so easy for `#[pyproto]`, in which case I'd be fine with leaving `#[pyproto]` broken. We're deprecating it anyway 😁